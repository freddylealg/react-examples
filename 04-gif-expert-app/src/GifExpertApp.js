import { useState} from 'react'
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState( defaultCategories );

    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
            { categories.map( (category, i ) => {
                return <GifGrid category={category} key={category}/>;
                }) 
            }
            </ol>
        </div>
        );
}


export default GifExpertApp;