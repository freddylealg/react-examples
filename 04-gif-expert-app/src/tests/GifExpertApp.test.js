import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';
import toJson from 'enzyme-to-json';
import GifExpertApp from '../GifExpertApp';

describe('Prueba el componente GifExpertApp', () => {

    test('Debe mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('Debe mostrar una lista de categorias', () => {
        const categories = ['Programing', 'Vegeta']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    })
    
})
