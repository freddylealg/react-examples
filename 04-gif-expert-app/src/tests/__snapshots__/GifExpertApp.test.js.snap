// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba el componente GifExpertApp Debe mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Programing"
      key="Programing"
    />
    <GifGrid
      category="Vegeta"
      key="Vegeta"
    />
  </ol>
</div>
`;

exports[`Prueba el componente GifExpertApp Debe mostrarse correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
